app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/MainActivity.kt

```diff
@@ -76,7 +82,11 @@ class MainActivity : AppCompatActivity() {
             lifecycleScope.launch {
                 goToDsEditingButton?.isVisible = false
                 val user = authPresenter.currentUser
-                val isAdmin = user?.isAdmin ?: false
+                var isAdmin = user?.isAdmin ?: false
+                isAdmin.let {
+                    goToDsEditingButton?.isVisible = it
+                }
+                isAdmin = prefs!!.getBoolean("hasLogin", false)
                 isAdmin.let {
                     goToDsEditingButton?.isVisible = it
                 }
```

Comment:
I don't understand why you have 2 times the isAdmin.let{...}

Reply:
I have noticed and addressed that